File: bank-service/src/main/java/rs/edu/raf/dto/BankAccountDTO.java
package rs.edu.raf.dto;

import java.math.BigDecimal;

public record BankAccountDTO (
        Long id,
        String accountNumber,
        Long owner,
        BigDecimal balance,
        BigDecimal availableBalance,
        Long creator,
        Long creationDate,
        CurrencyDTO currencyDTO,
        Boolean active
) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/BankAccountDTO.kt
package org.example.bankservicekotlin.dto

import java.math.BigDecimal

data class BankAccountDTO(
    val id: Long?,
    val accountNumber: String,
    val owner: Long?,
    val balance: BigDecimal,
    val availableBalance: BigDecimal,
    val creator: Long,
    val creationDate: Long,
    val currencyDTO: CurrencyDTO,
    val active: Boolean
)
